src/components/BBCode/BBCode.tsx

```diff
@@ -1,15 +1,77 @@
-import React, { FC } from 'react';
-
-import BBCodeUntyped from '@bbob/react/es/Component';
-import reactPreset from '@bbob/preset-react/es';
-
-import cls from './BBCode.module.css';
-
-/**
- * Typed BBCode component.
- */
-export const BBCode: FC = ({ children }) => (
-  <BBCodeUntyped plugins={[reactPreset()]} componentProps={{ className: cls.bbcode }}>
-    {children}
-  </BBCodeUntyped>
-);
+import React, { ReactElement, VFC } from 'react';
+
+import { parse, ast, ast_item, ast_to_array } from '@prekel/rescript-bbcode/src/BBCode.gen';
+import { Box, Link, Text } from '@chakra-ui/react';
+import cls from './BBCode.module.css';
+
+/**
+ * Transforms AST to React element array through astItemToElement function.
+ * @param parsed Parsed representation of BBCode-text (AST).
+ */
+const mapAstToElements = (parsed: ast): ReactElement[] => {

```

Comment:
I think 'parsed' is strange name for function param (not only here). Pls use more specific name.